src/main/scala/io/iohk/ethereum/blockchain/sync/FastSync.scala

```diff
@@ -112,6 +112,10 @@ class FastSync(
     private var requestedBlockBodies: Map[ActorRef, Seq[ByteString]] = Map.empty
     private var requestedReceipts: Map[ActorRef, Seq[ByteString]] = Map.empty
 
+
+    private val MAXIMUM_CACHE_SIZE = 500 // todo adjust cache size
+    private val blockSyncCache = CacheBuilder.newBuilder().maximumSize(MAXIMUM_CACHE_SIZE).build[ByteString, (Option[BlockHeader], Boolean, Boolean)]()
```

Comment:
Maybe you could also put `TotalParentDifficulty` in cache, then when geting it https://github.com/input-output-hk/mantis/pull/535/files#diff-3766981368048abce715f8396a3aed8cR305 it coult be taken form cache insted db ? (always one read less)